base_domain_operator/tests/test_base_domain_operator.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp.tests.common import TransactionCase
+
+
+class TestBaseDomainOperator(TransactionCase):
+    def test_base_domain_operator(self):
+        self.env['base.domain.operator']._register_hook()
+        # parent_of with an id
+        child = self.env.ref('base.res_partner_main1')
```

Comment:
I prefer not to depend on demo data as this tends to change a lot over versions and forces you to check new XML-IDs, if the data are the same...

It's very simple to create a partner hierarchy here.